### DISCUSSION TITLE
Set the default value of search bar?

### DISCUSSION THREAD

**Discussion by EvanBacon** [software-mansion/react-native-screens#1617] [Q&A]
Is it possible to set the default value of the search bar? Something like:

```diff
{
  headerSearchBarOptions: {
    placeholder: "Search",
    autoFocus: true,
+    value: 'incoming query'
  }
}
```

**Comment by felixaa** [comment]
I see that UISearchController has a method called `setText`

Would that be something that could be used?

```
- (void)setValue:(NSString *)value
{
  [_controller.searchBar setText:value];
}
```

My objc knowledge is limited but I tested the `setText` and it seems to work. But one would have to make sure it's emitting the 
onChangeText event, together with focusing the input

![Simulator Screen Recording - iPhone 14 Pro - 2022-12-02 at 12 51 25](https://user-images.githubusercontent.com/5445088/205286890-7a0956b4-ed85-4ec9-b1cb-7b94c1baac22.gif)

```
  [self emitOnSearchButtonPressEventWithText:text];
  [self showCancelButton];
  [self becomeFirstResponder];
  [self emitOnFocusEvent];
```

Maybe it's something that could be included with the imperative api? #1523

**Comment by tboba** [comment]
Hi @EvanBacon! Sorry for the late response :pray:
Currently you should have an ability to do something similar by using passing a ref to the `searchBar` option and using `setText(value)` method imperatively from it.
Let me know if that way of setting the value meets your needs.

Cheers!

**Comment by Rezrazi** [comment]
Hey @tboba 
I'm trying to implement a similar thing, but the ref doesn't seem to work as expected

```tsx
export default function Search() {
  const ref = useRef<SearchBarCommands>(null)

  useEffect(() => {
    console.log({ ref: ref.current })

    ref.current?.setText("Hello world")
  }, [])

  return (
    <>
      <Stack.Screen options={{
          title: "Search", headerSearchBarOptions: {
            placeholder: "Type search",
            ref: ref,
          },
        }}
      />
      
      <Text>Hello world</Text>
    </>
  )
}
``` 

The value doesn't get set and the ref is null in the log.

any insights?

**Comment by ipoogleduck** [comment]
Also looking for an answer to this

**Comment by ipoogleduck** [comment]
Ok I got it working by looking at apps/src/tests/Test1097.tsx in this repository. You can set the ref like this:
<img width="548" alt="Screenshot 2024-11-12 at 4 39 28 PM" src="https://github.com/user-attachments/assets/8474a6a9-91e2-4176-91dd-95de485c6ca9">
And then use any of the following functions:
<img width="367" alt="Screenshot 2024-11-12 at 4 39 36 PM" src="https://github.com/user-attachments/assets/8adbef78-51f8-4cff-81b8-207c6e33ab90">
Would be useful if this was in the documentation
